org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/buildutils/FilterablePathTraverser.xtend

```diff
@@ -0,0 +1,42 @@
+package org.testeditor.web.xtext.index.buildutils
+
+import com.google.common.base.Predicate
+import com.google.common.collect.Sets
+import java.io.File
+import java.io.FileFilter
+import org.eclipse.emf.common.util.URI
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor
+import org.eclipse.xtext.mwe.PathTraverser
+
+/**
+ * Traverses only directories and files that are accepted by the provided filter.
```

Comment:
a bit misleading isn't it? directories are not checked against the predicate, only files are. is this the intention?

Reply:
Actually, that's not true as far as I understand it. Maybe there's a misunderstanding of `java.io.File.listFiles`' semantics? Because that method returns all direct children of a directory, including its sub-directories, i.e. its a list of files _and_ folders.

You may also want to take a look at [the corresponding test](https://github.com/test-editor/xtext-dropwizard/blob/af6047fc071e5dd8d8b95d746885c1cfa7b14b2f/org.testeditor.web.dropwizard.xtext/src/test/java/org/testeditor/web/xtext/index/buildutils/FilterablePathTraverserTest.xtend), which should demonstrate that directories are indeed filtered out, unless I'm completely missing something here.

Reply:
what I meant was that the predicate is not applied on the directory itself. listFiles is executed even if the directory would no pass the predicate check.